packages/block-editor/src/components/block-lock/toolbar.js

```diff
@@ -49,7 +63,7 @@ export default function BlockLockToolbar( { clientId, wrapperRef } ) {
 				?.focus();
 		}
 		// wrapperRef is a reference object and should be stable
-	}, [ isModalOpen, shouldHideBlockLockUI, wrapperRef ] );
+	}, [ isModalOpen, shouldHideBlockLockUI, hasModalOpened, wrapperRef ] );
```

Comment:
Since `hasModalOpened` is a ref, I don't think it should be listed as a `useEffect` dependency